src/model/CoilWaterHeatingDesuperheater.cpp

```diff
@@ -498,15 +517,20 @@ namespace model {
     : StraightComponent(CoilWaterHeatingDesuperheater::iddObjectType(), model) {
     OS_ASSERT(getImpl<detail::CoilWaterHeatingDesuperheater_Impl>());
 
+    // That's the only bit that's user-specified and could be expected to really throw
+    bool ok = setSetpointTemperatureSchedule(setpointTemperatureSchedule);
+    if (!ok) {
+      remove();
+      LOG_AND_THROW("Unable to set " << briefDescription() << "'s Setpoint Temperature Schedule to " << setpointTemperatureSchedule.briefDescription()
+                                     << ".");
+    }
+
+    // All of the below is expected to work so an OS_ASSERT is good enough
     Schedule availabilitySchedule = model.alwaysOnDiscreteSchedule();
-    bool ok = setAvailabilitySchedule(availabilitySchedule);
-    OS_ASSERT(ok);
-    ok = setSetpointTemperatureSchedule(setpointTemperatureSchedule);
+    ok = setAvailabilitySchedule(availabilitySchedule);
     OS_ASSERT(ok);
     ok = setDeadBandTemperatureDifference(5.0);
     OS_ASSERT(ok);
-    ok = setRatedHeatReclaimRecoveryEfficiency(0.8);
-    OS_ASSERT(ok);
```

Comment:
Don't hard set a field that we have made optional.

Reply:
But this field _is_ required though, right? Meaning, the desuperheater requires a rated heat reclaim efficiency value for any heating source? If true, this seems a little weird. Perhaps the IDD field _should_ be required?

Reply:
It is weird, but that's because E+ will internally use a default based on the source. https://bigladdersoftware.com/epx/docs/22-1/input-output-reference/group-heating-and-cooling-coils.html#field-rated-heat-reclaim-recovery-efficiency